std/fmt.zig

```diff
@@ -20,17 +36,28 @@ pub fn format(
     comptime fmt: []const u8,
     args: ...,
 ) Errors!void {
+    if (args.len > 32) {
+        @compileError("32 arguments max are supported per format call");
+    }
+
     const State = enum {
         Start,
-        OpenBrace,
+        Positional,
         CloseBrace,
-        FormatString,
+        Specifier,
+        FormatWidth,
+        FormatPrecision,
         Pointer,
     };
 
     comptime var start_index = 0;
     comptime var state = State.Start;
     comptime var next_arg = 0;
+    comptime var maybe_pos_arg: ?comptime_int = null;
```

Comment:
May be tempted to move all this state into a struct if the number of fields here grow. Still need to add a few extra features (fill, alignment etc).